docs/debugging.md

```diff
@@ -0,0 +1,90 @@
+# Debugging Gate Failures
+
+## Blue Ocean
+[Blue ocean](https://rpc.jenkins.cit.rackspace.net/blue/organizations/jenkins/pipelines) is the new UI for [Jenkins](https://rpc.jenkins.cit.rackspace.net/), it makes it easier to find the point where
+a job failed, without trawling through the console. To use blue ocean, follow
+the link from a github pull request status to a job, then click open "Open Blue
+Ocean" from the left sidebar.
+
+<img src="images/debugging/sidebar.png" width=200>
+
+Blue ocean displays a horizontal pipeline of stages. Stages that failed will
+be shown in red. Clicking a stage reveals a list of steps beneath,
+steps that failed will also be marked as red. Expanding a step will show it's logs.
+
+![](images/debugging/blueocean.png)
+
+Using this method is much quicker for finding the failure point than reading the
+console.
+
+## The Console
+Jenkins collects all the output of commands it executes, these outputs are
+streamed to the console for each Job. This includes logging for Jenkins
+operations such as creating a single use instance for a project's jobs to
+execute on. As the console includes logging for steps not defined by a project,
+its inefficient to wade through it when looking for the cause of a failure.
+
+## Artefacts
+While command outputs as viewed in blue ocean or the console are useful, they
+don't tell the whole story. Jobs can produce artefacts which are stored
+for a [set time](https://github.com/rcbops/rpc-gating/blob/master/playbooks/upload_to_cloud_files.yml#L13-L19) after the completion of a job. For RPC-O AIO builds, these
+artefacts include the etc and log dirs of each container and the host.
+
```

Comment:
Given that at some point the artefacts that get stored for a particular job type should be definable by the team that is responsible for the software, maybe we should genericise this section and add a separate note to indicate RPCO specifics (which we will remove once we get to the point that teams define their own artefacts)